src/Microsoft.AspNetCore.Razor.Evolution/Intermediate/ExtensionIRNode.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Razor.Evolution.CodeGeneration;
+
+namespace Microsoft.AspNetCore.Razor.Evolution.Intermediate
+{
+    public abstract class ExtensionIRNode : RazorIRNode
+    {
+        internal abstract void WriteNode(RuntimeTarget target, CSharpRenderingContext context);
```

Comment:
This is internal right now because the rendering context is internal. We'll need a follow up to polish up this API and finalize it.